IdentityCore/src/cache/token/MSIDCredentialCacheItem.m

```diff
@@ -239,7 +243,8 @@ - (BOOL)matchesWithHomeAccountId:(nullable NSString *)homeAccountId
                      environment:(nullable NSString *)environment
               environmentAliases:(nullable NSArray<NSString *> *)environmentAliases
 {
-    if (homeAccountId && ![self.homeAccountId isEqualToString:homeAccountId])
+    if (homeAccountId && 
```

Comment:
Aren't all those fields already trimmed when the key is created?
I would like to avoid double trimming/lowercasing. 
Additionally, consider adding a category for NSString to do both of those operations to make the code more readable.

Reply:
added `msidNormalizedString`

Reply:
"Aren't all those fields already trimmed when the key is created?" - nope, they're not. I have a test failing. If they're supposed to, feel free to stop by and we can dig into why it's not happening